### PR TITLE
non-cURL, SSL warning, etc.

### PR DESCRIPTION
**Pull request by ngfeldman**
Got non-cURL api calls to work, made social publisher be auto-disabled and error message for when SSL not enabled, and fixed small bug in base_facebook for when api calls don't go through and return 'http_failure' as code instead of an int

### CODE REVIEW

**Comment by ngfeldman** [comment]
sorry for putting this all into one PR, i'll do a better job of using separate branches and keeping PRs to a single topic in the future

**Comment by mattwkelly** [comment]
I'm getting this when I land on the Facebook settings page.

---

No issues here!
Notice: fopen(): Content-type not specified assuming application/x-www-form-urlencoded in /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/class-http.php on line 877

Call Stack:
    0.0013     411588   1. {main}() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-admin/admin.php:0
    0.9705   23500144   2. do_action() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-admin/admin.php:151
    0.9707   23501324   3. call_user_func_array() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/plugin.php:405
    0.9707   23501352   4. fb_settings_page() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/plugin.php:405
    1.0152   23507264   5. fb_get_social_publisher_fields() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/fb-admin-menu.php:355
    1.0152   23507428   6. fb_get_user_pages() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/fb-social-publisher.php:528
    1.0152   23508096   7. WP_BaseFacebook->api() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/fb-social-publisher.php:695
    1.0154   23509228   8. call_user_func_array() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/includes/facebook-php-sdk/base_facebook.php:613
    1.0154   23509608   9. WP_BaseFacebook->_graph() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/includes/facebook-php-sdk/base_facebook.php:613
    1.0156   23510088  10. WP_BaseFacebook->_oauthRequest() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/includes/facebook-php-sdk/base_facebook.php:832
    1.0156   23510088  11. WP_BaseFacebook->getAccessToken() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/includes/facebook-php-sdk/base_facebook.php:853
    1.0156   23510208  12. WP_BaseFacebook->getUserAccessToken() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/includes/facebook-php-sdk/base_facebook.php:377
    1.0157   23510280  13. WP_BaseFacebook->getAccessTokenFromCode() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/includes/facebook-php-sdk/base_facebook.php:411
    1.0159   23510844  14. WP_BaseFacebook->_oauthRequest() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/includes/facebook-php-sdk/base_facebook.php:740
    1.0161   23511448  15. Facebook_WP_Extend->makeRequest() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/includes/facebook-php-sdk/base_facebook.php:863
    1.0179   23515696  16. wp_remote_post() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/includes/facebook-php-sdk/class-facebook-wp.php:41
    1.0179   23515696  17. WP_Http->post() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/http.php:92
    1.0181   23516724  18. WP_Http->request() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/class-http.php:280
    1.0206   23522352  19. WP_Http->_dispatch_request() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/class-http.php:192
    1.0209   23522712  20. WP_Http_Streams->request() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/class-http.php:255
    1.0212   23530788  21. fopen() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/class-http.php:877

Notice: fopen(): Content-type not specified assuming application/x-www-form-urlencoded in /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/class-http.php on line 877

Call Stack:
    0.0013     411588   1. {main}() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-admin/admin.php:0
    0.9705   23500144   2. do_action() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-admin/admin.php:151
    0.9707   23501324   3. call_user_func_array() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/plugin.php:405
    0.9707   23501352   4. fb_settings_page() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/plugin.php:405
    1.0152   23507264   5. fb_get_social_publisher_fields() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/fb-admin-menu.php:355
    1.0152   23507428   6. fb_get_user_pages() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/fb-social-publisher.php:528
    1.0152   23508096   7. WP_BaseFacebook->api() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/fb-social-publisher.php:695
    1.0154   23509228   8. call_user_func_array() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/includes/facebook-php-sdk/base_facebook.php:613
    1.0154   23509608   9. WP_BaseFacebook->_graph() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/includes/facebook-php-sdk/base_facebook.php:613
    1.0156   23510088  10. WP_BaseFacebook->_oauthRequest() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/includes/facebook-php-sdk/base_facebook.php:832
    1.1387   23512552  11. Facebook_WP_Extend->makeRequest() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/includes/facebook-php-sdk/base_facebook.php:863
    1.1403   23514724  12. wp_remote_post() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-content/plugins/facebook/includes/facebook-php-sdk/class-facebook-wp.php:41
    1.1403   23514724  13. WP_Http->post() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/http.php:92
    1.1404   23515752  14. WP_Http->request() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/class-http.php:280
    1.1434   23521376  15. WP_Http->_dispatch_request() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/class-http.php:192
    1.1437   23521460  16. WP_Http_Streams->request() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/class-http.php:255
    1.1442   23529444  17. fopen() /srv/www/mattwkelly.com/public_html/fb-hacks/wp/wp-includes/class-http.php:877

**Comment by mattwkelly** [comment]
I'm also getting it when I try to add a new post.

File: includes/facebook-php-sdk/class-facebook-wp.php [hunk 1]
```diff
@@ -33,7 +33,7 @@ protected function makeRequest( $url, $params, $ch=null ) {
 			'httpversion' => '1.1',
 			'timeout' => 60,
 			'user-agent' => apply_filters( 'http_headers_useragent', 'WordPress/' . $wp_version . '; ' . get_bloginfo( 'url' ) . '; facebook-php-' . self::VERSION . '-wp' ),
-			'headers' => array( 'Expect:' ),
+			'headers' => array( 'Connection' => 'close'),
```
Comment (ngfeldman) [inline includes/facebook-php-sdk/class-facebook-wp.php:36]: @mattwkelly Try replacing this line with
'headers' => array( 'Connection' => 'close' , 'Content-type' => 'application/x-www-form-urlencoded'),
and I think your weird message should disappear.
It looks like your server is complaining about a missing HTTP header and assuming this one. I don't know what's different about yours that it feels the need to complain.

Comment (mattwkelly) [inline includes/facebook-php-sdk/class-facebook-wp.php:36]: Fixed it.

'headers' => array( 'Connection' => 'close' , 'Content-type' => 'application/x-www-form-urlencoded'),

**Pull request merged**